test/units/module_utils/common/validation/test_check_type_int.py

```diff
@@ -31,4 +31,4 @@ def test_check_type_int_fail():
     for case in test_cases:
         with pytest.raises(TypeError) as e:
             check_type_int(case)
-    assert 'cannot be converted to an int' in to_native(e)
+        assert 'cannot be converted to an int' in to_native(e)
```

Comment:
Just a note, we probably should migrate these checks to use ```pytest.raises(TypeError, match='cannot be converted to an int')```  I had been hesitant to use that because I remembered having issues with it.  Turns out that it was a different parameter, ```message``` which should never be used.  (pytest has deprecated message because the same issue affected many people https://docs.pytest.org/en/latest/reference.html#pytest-raises )

I recommend it when looking at this code because I notice that we're doing ```to_native(e)``` as if e was an exception object here.   It's actually a pytest container that holds the actual exception inside of it.  Getting the str representation of the pytest container seems to return the string representation of the exception for now but it's not documented in pytest so it might not work in the future.

Reply:
heh :-)  I see that you found this out here: https://github.com/ansible/ansible/pull/60246 ;-)